src/cgnslib.c

```diff
@@ -833,6 +833,14 @@ int cg_get_compress(int *compress)
     return CG_OK;
 }
 
+int cg_set_filter(const struct CG_FILTER *filter)
```

Comment:
@MicK7 Do you think it is ok to use a struct to pass the filter API parameters? It does not look like it is typical to use a struct in the CGNS APIs. 
Otherwise, I have to pack the parameters into an array (with type conversion) for cg_configure and then unpack the array with type conversion.

Reply:
A struct seems usable in this case, and I don't really like the type conversion stuff as it is error prone. Maybe a typedef is necessary to make the code more CGNS  like.